### PR TITLE
Fix twitter integration

### PR DESCRIPTION
**Pull request by vgoh1**
# Description 
Twitter integration was broken because when I refactored it over the break I didn't test this specific endpoint. This is the fix. Also added validation for the input and a loading spinner since it's around 1s where nothing is happening as you're fetching the Twitter data.

# Type of change 
- [x] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 

# Problem
See description.

# Solution description
Validation using formik and yup (thanks Mable)
You don't have to click to show the input box now since I thought it was unnecessary.
Added loading spinner

# Tests 
**You'll need the twitter bearer token to test. DM me**
These cases should be handled in the input box for the twitter handle:
- one or more spaces 
- a valid twitter account
- a twitter account that doesn't exist
- more than 15 chars
- empty string
Feel free to test for more.

# Relevant document/ link (if any) 
N/A

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low

### CODE REVIEW

File: client/src/actions/team.js [hunk 1]
```diff
@@ -94,9 +98,11 @@ export const linkTwitter = (teamId, handle) => async (dispatch) => {
  */
 export const unlinkTwitter = (teamId) => async (dispatch) => {
   try {
+    console.log('unlink twitter');
     const { data } = await api.deregisterTwitterHandle(teamId, {
       twitterHandle: '',
     });
+    console.log(data);
```
Comment (aksh0001) [inline client/src/actions/team.js:105]: Prefer not to leave raw logs in main.

**Review by pmyip3 — APPROVED** [review]

File: client/src/actions/team.js [hunk 2]
```diff
@@ -68,6 +69,9 @@ export const getTeamInfo = (teamId) => async (dispatch) => {
  */
 export const linkTwitter = (teamId, handle) => async (dispatch) => {
   try {
+    dispatch({
+      type: FETCH_TWITTER_INFO,
+    });
```
Comment (aksh0001) [inline client/src/actions/team.js:74]: Please remove if not being used.

File: client/src/components/twitter/TwitterLink.js [hunk 3]
```diff
@@ -1,68 +1,77 @@
...
-import Jumbotron from 'react-bootstrap/Jumbotron';
-import Form from 'react-bootstrap/Form';
-import Col from 'react-bootstrap/Col';
 
 const TwitterLink = () => {
   const dispatch = useDispatch();
-  const [handle, setHandle] = useState('');
-  const [revealInput, setRevealInput] = useState(false);
   const teamId = useSelector((state) => state.team.teamId);
+  const [loadingState, setLoadingState] = useState(false);
```
Comment (aksh0001) [inline client/src/components/twitter/TwitterLink.js:17]: nit: prefer calling this just `isLoading`, or equivalent; i.e. remove the `state` suffix as its implied.

File: client/src/reducers/teamReducer.js [hunk 4]
```diff
@@ -33,15 +34,21 @@ const INITIAL_TEAM_STATE = {
...
+        ...state,
+        teamId: action.payload._id,
         email: action.payload.email,
-        teamName: action.payload.teamName, 
+        teamName: action.payload.teamName,
         orgName: action.payload.orgName,
         twitterHandle: action.payload.twitterHandle,
       };
+    case FETCH_TWITTER_INFO:
+      return { ...state, twitterFetchLoading: true };
```
Comment (aksh0001) [inline client/src/reducers/teamReducer.js:46]: Same, remove if unused.

**Review by aksh0001 — CHANGES_REQUESTED** [review]
Could you also address the SA checks for this file?
`client/src/components/layouts/DashboardSidebarData.js`

**Comment by vgoh1** [comment]
@aksh0001 re that SA warning: it's weird because BsUpload is being used in the file....so not sure why it's not picking that up. I think it's being removed in Jie's PR anyway so it should go away when that gets merged in.

File: client/src/actions/team.js [hunk 5]
```diff
@@ -95,6 +95,7 @@ export const linkTwitter = (teamId, handle) => async (dispatch) => {
  */
 export const unlinkTwitter = (teamId) => async (dispatch) => {
   try {
+    console.log('unlink twitter');
```
Comment (aksh0001) [inline client/src/actions/team.js:98]: nit: remove log

**Review by aksh0001 — APPROVED** [review]

**Review by rpra0006 — APPROVED** [review]
Looks good

**Pull request merged**